hera_cal/tempcal.py

```diff
@@ -0,0 +1,203 @@
+
+        # get keys
+        if keys is None:
+            keys = list(data.keys())
+
+        # smooth data
+        if not hasattr(self, 'smooth'):
+            self.smooth = DataContainer({})
+        if not hasattr(self, 'ratio'):
+            self.ratio = DataContainer({})
+        for k in keys:
+            # smooth data
+            utils.echo("working on {}".format(k), verbose=verbose)
+            gkey = utils.split_bl(k)[0]
+            data_shape = data[k].shape 
+            self.smooth[k] = utils.gp_interp1d(times, data[k], flags=flags[k], length_scale=length_scale, nl=nl,
+                                               Nmirror=Nmirror)
+            self.gflags[gkey] = np.isclose(self.smooth[k], 0.0)
+
+            # take ratio and compute gain term
+            self.ratio[k] = np.true_divide(self.data[k], self.smooth[k], where=~self.gflags[gkey])
+            self.gains[gkey] = np.sqrt(self.ratio[k])
+
+            # only allow frequency-averaged gains for now via nanmedian
+            # use median over mean to help w/ unflagged RFI
+            g = self.gains[gkey].copy()
+            g[self.gflags[gkey]] = np.nan
+            self.gains[gkey] = np.nanmedian(g, axis=1, keepdims=True)
+            self.gains[gkey][np.isnan(self.gains[gkey])] = 1.0  # catch for fully flagged integrations
+            self.gains[gkey] = np.repeat(self.gains[gkey], data_shape[1], axis=1)
+            self.gflags[gkey] = np.repeat(np.min(self.gflags[gkey], axis=1, keepdims=True), data_shape[1], axis=1)
+
+    def avg_ants(self, avg_ants):
+        """
+        Average ant-pol keys together in self.gains.
+
+        Args:
+            avg_ants : list of tuple or list of list of tuples
+                List of ant-pol tuples to average together.
+
+        Result:
+            Averaged gains edited inplace in self.gains and self.gflags.
+        """
+        # average antennas
+        if avg_ants is not None:
+            # get gain keys
+            keys = list(self.gains.keys())
+            assert isinstance(avg_ants, list)
+            if not isinstance(avg_ants[0], list):
+                avg_ants = [avg_ants]
+            # iterate over antenna lists
+            for antlist in avg_ants:
+                gkeys = [k for k in antlist if k in self.gains]
+                avg = np.sum([self.gains[k] * ~self.gflags[k] for k in gkeys], axis=0) / np.sum([~self.gflags[k] for k in gkeys], axis=0).clip(1e-10, np.inf)
+                avgf = np.min([self.gflags[k] for k in gkeys], axis=0)
+                for gkey in gkeys:
+                    self.gains[gkey] = avg
+                    self.gflags[gkey] = avgf
+
+    def set_abscal_time(self, times, cal_time):
+        """
+        Normalize gains by Julian Date of abscal.
+
+        Args:
+            times : ndarray
+                Times of data in Julian Date.
+            cal_time : float
+                Julian Date of when absolute calibration was performed.
```

Comment:
This is confusing. This really has nothing to do with abscal. This function is way more general than that. It's just pick a time to call |g| = 1. Could be anything! I'd rename the arguments and function to clarify that it's just `normalize_gains_at_given_time` or something like that.

Reply:
I mean, in practice it has everything to do with abscal: that's exactly why you'd use this function. It can be a general function, but it may be useful to keep its use specific to why we'd use it _in the context of temperature calibration_. If we had a function say in `utils` that did this, it might be named as you suggest.

Reply:
I don't think it needs to move to utils until enough other code wants to use it, but giving it a very specific name when it's a more general function means that we're more likely to rewrite the code than reuse it if we ever have to do it again in another context. I'm totally fine with the docstring highlighting the specific use case.

Reply:
but I think we should be specific if its in `TempCal`: if we want to use this functionality we shouldn't be calling it from `tempcal.TempCal`. I feel that if you want to rename it to something general, let's move it to utils, otherwise let's keep it as is